File: 1186-maximum-subarray-sum-with-one-deletion/1186-maximum-subarray-sum-with-one-deletion.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var maximumSum = function(arr) {
    let keep = arr[0];
    let del = -Infinity;
    let ans = arr[0];

    for (let i = 1; i < arr.length; i++) {
        let prevKeep = keep;
        let prevDel = del;

        keep = Math.max(prevKeep + arr[i], arr[i]);
        del = Math.max(prevKeep, prevDel + arr[i]);
        ans = Math.max(keep, del, ans);
    }
    return ans;
};

File: 1186-maximum-subarray-sum-with-one-deletion/1186-maximum-subarray-sum-with-one-deletion.go
func maximumSum(arr []int) int {
	keep := arr[0];
	del := -1000000000;
	ans := arr[0];

	for i := 1; i < len(arr); i++ {
		prevKeep := keep;
		prevDel := del;

		keep = max(prevKeep+arr[i], arr[i]);
		del = max(prevDel+arr[i], prevKeep);
		ans = max(keep, del, ans);
	}
	return ans;
}